docs/io/votable/dataorigin.rst

```diff
@@ -0,0 +1,106 @@
+
+Introduction
+------------
+Extract basic provenance information from VOTable header. The information is described in
+DataOrigin IVOA note: https://www.ivoa.net/documents/DataOrigin/.
+
+DataOrigin includes both the query information (such as publisher, contact, versions, etc.) 
+and the Dataset origin (such as Creator, bibliographic links, URL, etc.)
+
+This API retrieves Metadata from INFO in VOTable.
+
+
+Getting Started
+---------------
+
+To extract DataOrigin from VOTable
+
+Example: VizieR catalogue J/AJ/167/18
+
+.. code-block:: python
+
+    >>> from astropy.io.votable import parse
+    >>> from astropy.io.votable.dataorigin import extract_data_origin
+
+    >>> votable = parse("https://vizier.cds.unistra.fr/viz-bin/conesearch/J/AJ/167/18/table4?RA=265.51&DEC=-22.71&SR=0.1")  # doctest: +REMOTE_DATA
+    >>> data_origin = extract_data_origin(votable)  # doctest: +REMOTE_DATA
+    >>> print(data_origin)  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+    publisher: CDS
+    server_software: 7.4.5
+    service_protocol: ivo://ivoa.net/std/ConeSearch/v1.03
+    request_date: 2025-03-05T14:18:05
+    contact: cds-question@unistra.fr
+    publisher: CDS
+
+    ivoid: ivo://cds.vizier/j/aj/167/18
+    citation: doi:10.26093/cds/vizier.51670018
+    reference_url: https://cdsarc.cds.unistra.fr/viz-bin/cat/J/AJ/167/18
+    rights_uri: https://cds.unistra.fr/vizier-org/licences_vizier.html
+    creator: Hong K.
+    ...
+
+Contents and metadata
+---------------------
+
+`astropy.io.votable.dataorigin.extract_data_origin` returns a `astropy.io.votable.dataorigin.DataOrigin` (class) container which is made of:
+
+* a `astropy.io.votable.dataorigin.QueryOrigin` (class) container describing the request.
+  ``QueryOrigin`` is considered to be unique for the whole VOTable.
+  It includes metadata like  the publisher, the contact, date of execution, query, etc.
+
+*  a list of `astropy.io.votable.dataorigin.DatasetOrigin` (class) container for each Element having DataOrigin information.
+   ``DataSetOrigin`` is a basic provenance of the datasets queried. Each attribute is a list.
+   It includes metadata like authors, ivoid, landing pages, ....
+
+Examples
+--------
+
+Get the (Data Center) publisher and the Creator of the dataset
+
+.. code-block:: python
+
+    >>> print(data_origin.query.publisher)  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+    CDS
+    >>> print(data_origin.origin[0].creator)  # doctest: +REMOTE_DATA +IGNORE_OUTPUT
+    ['Hong K.']
+
+Other capabilities
+------------------
+DataOrigin container includes VO Elements:
+
+* Extract list of `astropy.io.votable.tree.Info`
+
+.. code-block:: python
+
+    >>> # get DataOrigin with the description of each INFO
+    >>> for dataset_origin in data_origin.origin:  # doctest: +REMOTE_DATA
+    ...    for info in dataset_origin.infos:   # doctest: +REMOTE_DATA
+    ...        print(f"{info.name}: {info.value} ({info.content})")   # doctest: +REMOTE_DATA +IGNORE_OUTPUT
```

Comment:
we should try and drop `+IGNORE_OUTPUT` in some of these tests, so they can test more than for basic crashes. E.g. I feel this one would be a safe one to drop (and ultimately once more tests are added in the module itself (see Tom's comment), then we could tune back on doctesting if it turns out to be too flaky